frontend/src/hooks/useChess.js

```diff
@@ -196,6 +299,9 @@ export const useChess = ({ onPlayerMove, onBotMove, lock, isPaused, config }) =>
   const onSquareClick = (square) => {
     setRightClickedSquares({});
     if (isPaused) return; // Disable piece movement if game is paused
+    if (gameMode === 'timed') {
+      if (isGameOver) return;
+    }
```

Comment:
I see this is common pattern that repeats itself some more times, where the nesting is unecessary, it would be great if you just reviewed the code and try to see if you can refactor it a bit for readability's sake